staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go

```diff
@@ -186,98 +196,153 @@ func validateStorageVersion(t *testing.T, ctc *conversionTestContext) {
-	versions := []string{"v1beta1", "v1beta2"}
+		// Create CRs at all storage versions
+		objNames := []string{}
+		for _, version := range versions {
+			ctc.setAndWaitStorageVersion(t, version)
 
-	// Create CRs at all storage versions
-	objNames := []string{}
-	for _, version := range versions {
-		ctc.setAndWaitStorageVersion(t, version)
+			name := "mixedstorage-stored-as-" + version
+			obj, err := clients[version].Create(newConversionMultiVersionFixture(ns, name, version), metav1.CreateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			objNames = append(objNames, obj.GetName())
+		}
 
-		name := "stored-at-" + version
-		obj, err := clients[version].Create(newConversionMultiVersionFixture(ns, name, version), metav1.CreateOptions{})
-		if err != nil {
-			t.Fatal(err)
+		// Ensure copies of an object have the same fields and values at each custom resource definition version regardless of storage version
+		for clientVersion, client := range clients {
+			t.Run(clientVersion, func(t *testing.T) {
+				o1, err := client.Get(objNames[0], metav1.GetOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				for _, objName := range objNames[1:] {
+					o2, err := client.Get(objName, metav1.GetOptions{})
+					if err != nil {
+						t.Fatal(err)
+					}
+
+					// ignore metadata for comparison purposes
+					delete(o1.Object, "metadata")
+					delete(o2.Object, "metadata")
+					if !reflect.DeepEqual(o1.Object, o2.Object) {
+						t.Errorf("Expected custom resource to be same regardless of which storage version is used to create, but got: %s", cmp.Diff(o1, o2))
+					}
+				}
+			})
 		}
-		objNames = append(objNames, obj.GetName())
 	}
+}
+
+func validateServed(t *testing.T, ctc *conversionTestContext) {
+	ns := ctc.namespace
 
-	// Ensure copies of an object have the same fields and values at each custom resource definition version regardless of storage version
-	for clientVersion, client := range clients {
-		t.Run(clientVersion, func(t *testing.T) {
-			o1, err := client.Get(objNames[0], metav1.GetOptions{})
+	for _, version := range ctc.crd.Spec.Versions {
+		t.Run(version.Name, func(t *testing.T) {
+			name := "served-" + version.Name
+			client := ctc.versionedClient(ns, version.Name)
+			obj, err := client.Create(newConversionMultiVersionFixture(ns, name, version.Name), metav1.CreateOptions{})
 			if err != nil {
 				t.Fatal(err)
 			}
-			for _, objName := range objNames[1:] {
-				o2, err := client.Get(objName, metav1.GetOptions{})
+			ctc.setServed(t, version.Name, false)
+			ctc.waitForServed(t, version.Name, false, client, obj)
+			ctc.setServed(t, version.Name, true)
+			ctc.waitForServed(t, version.Name, true, client, obj)
+		})
+	}
+}
+
+func validateNonTrivialConverted(t *testing.T, ctc *conversionTestContext) {
+	ns := ctc.namespace
+
+	for _, createVersion := range ctc.crd.Spec.Versions {
+		t.Run(fmt.Sprintf("getting objects created as %s", createVersion.Name), func(t *testing.T) {
```

Comment:
this gets used as the test name and should be more succinct (for use with -run, etc). suggest `nontrivial-converted-<version>`. (include in a follow-up PR)